lib/local_daily_digest.js

```diff
@@ -0,0 +1,167 @@
+var async = require('async'),
+	_ = require('lodash'),
+	log = require('../log'),
+	Profile = require('../models').Profile,
+	Contact = require('../models').Contact,
+	log = require('../log'),
+	mail = require('../mail');
+
+function get(req, res, next) {
+	var result = {},
+	 	allLocalProfilesWDigest = [],
+	 	allLocalProfiles =[],
+	 	requiredUsers=[],
+	 	checkedOutUsers = [],
+	 	finalCheckedOutList=[],
+		tempManagers =[],
+		tempUsers = [],
+		managerContactsProfile = [],
+		finalOutput = [],
+		tempProf=[],
+		requiredLocationsId =[],
+		requiredLocationsName =[],
+		temp='',
+		tempProfile ='',
+		checkedInUsers = [];
+	
+	async.series([
+		//THe local daily digest will be just the digest for that country !!
+		function (cb) {
+			Profile.find({'dailyDigest': {$not: {$size: 0}}}, function(err, profile){
+			if(err){
+		  		res.send(500, new Error(err));
+		  		return cb(true);
+		  	}
+
+		  	if(!profile){
+		  		res.send(400, new Error('No Profiles were found'));
+		  	}
+
+		  	profile.forEach(function(profile){
+			  		allLocalProfilesWDigest.push(profile);
+			  	});
+	  			return cb();
+			});
+		},
+		function(cb){
+			allLocalProfilesWDigest.forEach(function(profile){
+				// console.log(profile);
+				profile.dailyDigest.forEach(function(location){
+					requiredLocationsId.push(location);
+				});
+				// requiredLocationsName.push(profile.location);
+			});
+			console.log("the countries this profile is either a manger or editor of are: ", requiredLocationsId);
```

Comment:
Any check happening in the app are CONVENIENCE checks only: they help the user get immediate feedback about what they can do and can not do, BUT any check happening on the app should not be considered as secure. Any user with a big of technical knowledge can send ANY data to the backend (ie the profiles api) without going through the client app.
Checks MUST therefore happen server side, otherwise there will be a security issue or the risk of seeing wrong data stored in mongodb. Checks happening in the client app are only for better user experience.